monitoring/lifecycle/dashboard.py

```diff
@@ -0,0 +1,283 @@
+from grafanalib.core import (
+    BarGauge,
+    ConstantInput,
+    DataSourceInput,
+    GaugePanel,
+    Graph,
+    Heatmap,
+    HeatmapColor,
+    RowPanel,
+    Stat,
+    Threshold,
+    YAxis,
+)
+
+from grafanalib import formatunits as UNITS
+from scalgrafanalib import layout, Tooltip, Target, TimeSeries, Dashboard
+
+
+STATUS_CODE_2XX = '2xx'
+STATUS_CODE_3XX = '3xx'
+STATUS_CODE_4XX = '4xx'
+STATUS_CODE_5XX = '5xx'
+
+
+def s3_request_timeseries_expr(process, job, code):
+    labelSelector = 'namespace="${namespace}"'
+    labelSelector += f',status="{code}"'
+
+    if job is not None:
+        labelSelector += f',job="{job}"'
+
+    if process is not None:
+        labelSelector += f',origin="{process}"'
+
+    return f'sum(rate(lifecycle_s3_operations{{{labelSelector}}}[$__range]))'
+
+
+def s3_request_timeseries(title, process=None, job=None):
+    return TimeSeries(
+        title=title,
+        dataSource="${DS_PROMETHEUS}",
+        stacking={
+            "group": "A",
+            "mode": "normal",
+        },
+        targets=[
+            Target(
+                expr=s3_request_timeseries_expr(process, job, STATUS_CODE_2XX),
+                legendFormat="HTTP 2xx",
+            ),
+            Target(
+                expr=s3_request_timeseries_expr(process, job, STATUS_CODE_3XX),
+                legendFormat="HTTP 3xx",
+            ),
+            Target(
+                expr=s3_request_timeseries_expr(process, job, STATUS_CODE_4XX),
+                legendFormat="HTTP 4xx",
+            ),
+            Target(
+                expr=s3_request_timeseries_expr(process, job, STATUS_CODE_5XX),
+                legendFormat="HTTP 5xx",
+            ),
+        ]
+    )
+
+
+def s3_request_error_rate_expr(process, job, code):
+    divdLabel = 'namespace="${namespace}"'
+    divsLabel = 'namespace="${namespace}"'
+
+    if code is not None:
+        divdLabel += f',status="{code}"'
+    else:
+        divdLabel += ',status!="2xx"'
+
+    if job is not None:
+        divdLabel += f',job="{job}"'
+        divsLabel += f',job="{job}"'
+
+    if process is not None:
+        divdLabel += f',origin="{process}"'
+        divsLabel += f',origin="{process}"'
+
+    return f'sum(rate(lifecycle_s3_operations{{{divdLabel}}}[$__range]))/sum(rate(lifecycle_s3_operations{{{divsLabel}}}[$__range]))'
```

Comment:
to avoid "not a number", you need to filter the expression `>0` :

```
sum(rate(lifecycle_s3_operations{...}[$__range]))
  /  sum(rate(lifecycle_s3_operations{..}[$__range]) > 0)
```

Just found the issue here: https://scality.atlassian.net/browse/ARTESCA-3935

Reply:
(and use `$__rate_interval` ?)